### DISCUSSION TITLE
Composer install "Syncing my-private-repo bundle into cache" extrem slow

### DISCUSSION THREAD

**Discussion by mschweiger90** [composer/composer#11928] [Q&A / Support]
Hello everyone,

I have the following problem:
We use a gitlab-ci pipeline for a build process.
This executes a “composer install” command.

The following happens to all public repositories that are installed in this command:
"Downloading my-public-repo-bundle"

The following happens to all private repositories that are installed in this command:
"Syncing my-private-repo-bundle into cache"

The process “Syncing my-private-repo-bundle into cache” takes approx. 1-2 minutes per bundle!
Why is that? Or how can you prevent this?
The bundle from the private repo should not be synced into the cache.

A --no-cache on composer install does not change this behavior.

The private repositories are integrated into the Compser Json as follows:
`"repositories": {
    "0": {
      "type": "git",
      "url": "git@gitlab.com:my-private-bundle.git"
    },...`
    
Many thanks for your help!

**Comment by Seldaek** [comment]
That's because using git repo definitions like that you force cloning the whole repository to install the package. Using [Private Packagist](https://packagist.com) instead to sync those private repos into a single composer repository would make things much easier to use and faster to install.

**Comment by stof** [comment]
You should use the type `vcs` rather than `git`, which would also allow composer to use its special support of gitlab, allowing it to use the gitlab archives instead of cloning the whole repo.
